webextensions/api/webRequest.json

```
@@ -19,7 +19,8 @@
             },
             "opera": "mirror",
             "safari": {
-              "version_added": "14"
+              "version_added": "14",
+              "notes": "Support for use in Manifest V3 extensions, that is with non-persistent background pages/scripts, provided in Safari 18."
```

How about making the following, to be more explicit:

```suggestion
              "notes": [
                "Before Safari 18, the `webRequest` API was available for Manifest V2 extensions, and for Manifest V3 extensions only if they used non-persistent background content",
                "Since Safari 18, the `webRequest` API is available for both Manifest V2 and V3 extensions, including those with persistent background content."
              ]
```

---

@xeenon, however, isn't this potentially confusing, as MV3 doesn't support persistent background pages or scripts.
```suggestion
              "notes": [
                "Before Safari 18, the `webRequest` API could only be used by extensions with persistent background pages or scripts, so only supported Manifest V2 extensions.",
                "From Safari 18, the `webRequest` API is available for extensions using persistent or non-persistent  background pages or scripts, thus providing support for Manifest V2 and V3 extensions."
              ]
```

---

Good point. I think this can be simplified as:

Before Safari 18, `webRequest` only worked with Manifest V2 extensions using persistent background content. After Safari 18, it’s always available.

---

```suggestion
              "notes": "Before Safari 18, only worked with Manifest V2 extensions using persistent background content. From Safari 18, works with Manifest V2 and V3 extensions ."
```